### ISSUE TITLE
Please Add instructions for compiling SSD Mobilenet

### ISSUE THREAD

**Issue by mitchross** [dataplayer12/homesecurity#3]
It is expected that your Jetson TX2 already has numpy and opencv installed. Install the official tensorflow release for Jetson TX2 by following the instructions from here. You may also have to compile the MobileNet SSD model for your version of tensorflow and JetPack.

//TODO: Add instructions for compiling SSD Mobilenet

**Comment by dataplayer12** [comment]
Hi, I have added a new script in `jetsontx2` called `compile_ssd_mobilenet.py`. It will create an inference graph for the tensorflow version you are using. Please launch it from the root directory of the project.

```Shell
cd homesecurity/ #cd to project directory
python3 jetsontx2/compile_ssd_mobilenet.py
```
If it works, please close this issue.

**Comment by dataplayer12** [comment]
Hi, I believe the fix I have provided should work. I am closing this issue due to lack of activity.

**Comment by mitchross** [comment]
I'll give this a whirl tonight. I am substituting a jetson nano in place of a jetson tx2, can I still run this script?

**Comment by dataplayer12** [comment]
Yes, it should work, but you will get much lower performance than a TX2. I haven't tried it on a nano yet.

**Comment by mitchross** [comment]
I see an issue with the training script.

"from jetson_config" ->> ModuleNotFoundError: No module named 'jetson_config'

**Comment by dataplayer12** [comment]
Hi, thanks for pointing this out. I moved a lot of things around over the weekend and missed this script.

**Comment by dataplayer12** [comment]
I have now fixed the script. Please pull the latest commit.

**Comment by mitchross** [comment]
I know this project is for TX2, but this is a pretty helpful repo. For others that may come across on jetson nano they may want to learn how to get this working. Here is what I am seeing on Jetson Nano.

Traceback (most recent call last):
  File "compile_ssd_mobilenet.py", line 2, in <module>
    from tf_trt_models.detection import build_detection_graph
ModuleNotFoundError: No module named 'tf_trt_models'

**Comment by dataplayer12** [comment]
Yes, I forgot to mention that you should install a repo called `tf_trt_models`. This repo is here.

If you have tensorflow installed, please use only these commands:
```Shell
git clone --recursive https://github.com/NVIDIA-Jetson/tf_trt_models.git
cd tf_trt_models
./install.sh python3
```
I think I will just create an install script for users new to the jetson platform.

**Comment by mitchross** [comment]
Ok getting further, that script is helpful. I am still getting this error...

Traceback (most recent call last):
  File "compile_ssd_mobilenet.py", line 2, in <module>
    from tf_trt_models.detection import build_detection_graph
ModuleNotFoundError: No module named 'tf_trt_models'


If its helpful, this is the output of the install of the tf_trt_model script. ( I did do a source ~/.bashrc to ensure path variables are updated )

Processing tf_trt_models-0.0-py3.6.egg
Removing /home/vanillax/.local/lib/python3.6/site-packages/tf_trt_models-0.0-py3.6.egg
Copying tf_trt_models-0.0-py3.6.egg to /home/vanillax/.local/lib/python3.6/site-packages
tf-trt-models 0.0 is already the active version in easy-install.pth

Installed /home/vanillax/.local/lib/python3.6/site-packages/tf_trt_models-0.0-py3.6.egg
Processing dependencies for tf-trt-models==0.0
Finished processing dependencies for tf-trt-models==0.0

**Comment by mitchross** [comment]
I got a bit further. 

I couldnt run `git clone --recursive https://github.com/NVIDIA-Jetson/tf_trt_models.git
cd tf_trt_models
./install.sh python3`

as user, I had to run it with sudo. Not sure why, but I was getting permission denied if I didnt use sudo.

So then I ran sudo python3 compile_ssd_mobilenet.py  as root

I get 

`Traceback (most recent call last):
  File "compile_ssd_mobilenet.py", line 11, in <module>
    batch_size=1
  File "/home/vanillax/.local/lib/python3.6/site-packages/tf_trt_models-0.0-py3.6.egg/tf_trt_models/detection.py", line 118, in build_detection_graph
FileNotFoundError: [Errno 2] No such file or directory: './human_detection/ssd_mobilenet_v1_coco.confi`

**Comment by dataplayer12** [comment]
Are you calling `compile_ssd_mobilenet.py ` script with python3? 
```Shell
python3 compile_ssd_mobilenet.py #please do this
python compile_ssd_mobilenet.py #don't do this
```
Tensorflow support on jetson is usually better for python3.

**Comment by mitchross** [comment]
See above -

But when I ran the jetson tf_trt_model script as user I got

`error: [Errno 13] Permission denied: '/home/vanillax/.local/lib/python3.6'`

Not so much your issue as it is mine, but if you create a install script, then that may be a factor in your script. 

When I set up tensor flow, I used this guy's guide https://github.com/SteveMacenski/jetson_nano_detection_and_tracking

**Comment by dataplayer12** [comment]
Thank you for sharing the troubles you are having. I can offer two suggestions:
- Please pull the latest version of the repo.
- Please call the compile script from the base directory of the project, as
```Shell
cd homesecurity/
python3 jetsontx2/compile_ssd_mobilenet.py
```
All the scripts in the project should be called from the base directory because thats how all the paths have been configured.

**Comment by dataplayer12** [comment]
The guide you mentioned also has [this script](https://github.com/SteveMacenski/jetson_nano_detection_and_tracking/blob/master/tf_download_and_trt_model.py) which does basically the same thing as my compile script. If you can successfully run that script, you can just copy the `.pb` model file generated by that model to `jetsontx2` directory of this project.

**Comment by mitchross** [comment]
This worked! `python3 jetsontx2/compile_ssd_mobilenet.py`

Thank you for taking time to walk me through this. Hopefully this helps with your getting started readme. 

One note for suggestion. Take a RTSP stream as optional camera input. See https://harizanov.com/2018/03/enhancing-my-ordinary-security-cameras-with-ai/

I've been trying to combine many peoples projects into one that accepts Rpi cameras and RTSP streams from AI cameras to feed for inference. 

Then take a look at this commit for OpenCV support https://github.com/RazZziel/AI-ipcam/commit/a619f8170b9e32f4f911f70a8fd3f5a1be1666c3

I highly suggest taking a look at https://github.com/toolboc/IntelligentEdgeHOL for more ideas.

keep up the good work!

**Comment by dataplayer12** [comment]
Thank you for the ideas and links. 
- I plan to add support for RTSP streams. I had trouble getting them working on a pi zero, so I settled for a http stream for the time being.
- At some point, I would also like to see what NVIDIA's deep stream SDK can do to simplify/accelerate data streaming. It seems like using h264 streams can massively accelerate frame grabbing.
- Also on the cards is support for both tf_trt_models and darknet (yolo-v3). The xavier repository is the only one to use darknet at the moment, but I have verified that darknet works on all jetson devices.
- I also plan to make the code more flexible to be able to use more cameras. Jetson Xavier, especially can handle way more than 4 cameras.

I did not plan to add many features to this project, but it has gotten nice traction in the community, so I am willing to work more on this. Thanks for taking the time to point out errors and offer suggestions.
